### PR TITLE
Implement tx broadcast interface

### PR DESCRIPTION
**Pull request by andrei-21**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -21,28 +22,39 @@ use crate::keys_manager::{generate_secret, init_keys_manager};
...
+    esplora_client: Arc<AsyncClient>,
 }
 
 impl LightningNode {
     pub fn new(
         config: Config,
         redundant_storage_callback: Box<dyn RedundantStorageCallback>,
     ) -> Result<Self, InitializationError> {
-        let rt = AsyncRuntime::new()?;
+        let rt = Arc::new(AsyncRuntime::new()?);
```
Comment (danielgranhao) [inline src/lib.rs:45]: Instead of wrapping the runtime in an `Arc`, I think we can use a `tokio::runtime::Handle`.

**Review by danielgranhao — APPROVED** [review]
Only a minor detail. Otherwise LGTM.

**Comment by danielgranhao** [comment]
I've created an [issue](https://github.com/bitcoindevkit/rust-esplora-client/issues/11) to see if they would be open to allow the selection of the TLS implementation used by `reqwest`.

File: Cargo.toml [hunk 2]
```diff
@@ -17,8 +17,11 @@ thiserror = "1.0.34"
 tokio = { version = "1.21.1", features = ["rt-multi-thread", "time"] }
 uniffi = "0.20.0"
 
+esplora-client = { git = "https://github.com/bitcoindevkit/rust-esplora-client" }
```
Comment (gcomte) [inline Cargo.toml:20]: Specifying the async feature would safe us compilation time and create a smaller binary
`
esplora-client = { git = "https://github.com/bitcoindevkit/rust-esplora-client", default-features = false, features = ["async"] }`

**Comment by gcomte** [comment]
Clippy throws warnings ... (Our clippy CI check isn't working as intended yet)

File: Cargo.toml (hunk 2 above)
Comment (andrei-21) [reply Cargo.toml:20]: Good point, thanks.

File: src/lib.rs (hunk 1 above)
Comment (andrei-21) [reply src/lib.rs:45]: You are right, I will address it in a separate PR.

Comment (andrei-21) [reply src/lib.rs:45]: https://github.com/getlipa/lipa-lightning-lib/pull/33

**Review by danielgranhao — APPROVED** [review]

**Pull request merged**